archspec/cpu/microarchitecture.py

```diff
@@ -232,6 +239,11 @@ def optimization_flags(self, compiler, version):
             msg = msg.format(compiler, best_target, best_target.family)
             raise UnsupportedMicroarchitecture(msg)
 
+        # Check that the version matches the expected format
+        if not re.match(r"^(?:\d+\.)*\d+$", version):
+            msg = "wrong format for the version argument. Only dot separated digits are allowed."
```

Comment:
```suggestion
            msg = "Invalid format for archspec compiler version. Only dot separated digits are allowed."
```

Reply:
Two minor differences from suggestion:
1. I started the message with lowercase (following most Python built-in errors)
2. I didn't mention `archspec` since it is implied (but I added a mention to the "compiler")

Reply:
I think it's important to mention `archspec`, because we pass these error messages unchanged through other programs that have different requirements on their version syntax. Either that or use an Archspec-specific subclass of ValueError.

Reply:
See https://github.com/archspec/archspec/pull/155/commits/419c0084a1b06c09a2618dbd54b203f9a38fdbe5

Reply:
@becker33